registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,547 @@
+// Package swift provides a storagedriver.StorageDriver implementation to
+// store blobs in Openstack Swift object storage.
+//
+// This package leverages the ncw/swift client library for interfacing with
+// Swift.
+//
+// Because Swift is a key, value store the Stat call does not support last modification
+// time for directories (directories are an abstraction for key, value stores)
+package swift
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	gopath "path"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/ncw/swift"
+
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/base"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+)
+
+const driverName = "swift"
+
+const defaultChunkSize = 20 * 1024 * 1024
+
+const minChunkSize = 1 << 20
+
+const directoryMimeType = "application/directory"
```

Comment:
This may be insufficiently narrow for this use case. Maybe, make it clear that it is a vendor mimetype for a specific directory format for this particular use case.